applications/GeoMechanicsApplication/custom_utilities/generic_utilities.h

```diff
@@ -48,6 +49,36 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GenericUtilities
         return result;
     }
 
+    template <class ContainerType, class IdsType>
+    static void CollectIdsFromEntity(const ContainerType& rContainer, IdsType& rIds)
+    {
+        for (const auto& r_item : rContainer) {
+            const auto id = r_item.Id();
+
+            if constexpr (requires { rIds.insert(id); }) {
+                // Works for set/unordered_set
+                rIds.insert(id);
+            } else if constexpr (requires { rIds.push_back(id); }) {
+                // Works for vector/deque
+                rIds.push_back(id);
+            } else {
+                static_assert(!std::is_same_v<IdsType, IdsType>,
+                              "OutputContainerType does not support insert or push_back.");
+            }
+        }
+    }
+
+    template <class ContainerType>
+    static auto CollectIdsFromEntity(const ContainerType& rContainer)
+    {
+        using ItemRef = decltype(*std::begin(rContainer));
+        using Item    = std::remove_cv_t<std::remove_reference_t<ItemRef>>;
+        using IdType  = decltype(std::declval<Item>().Id());
+        std::vector<IdType> result;
```

Comment:
I think we can simplify this a bit, since the Id of something in Kratos is always a std::size_t
```suggestion
        std::vector<std::size_t> result;
```
I get that this will keep working if the id type ever gets changed, but since there are a lot of aliases already and I don't expect this to change, we can just use std::size_t in my opinion.

Reply:
changed. Indeed we return only a vector calling this function now.

Reply:
I agree with @rfaasse. Also note that the original code would lead to undefined behavior when an empty container is passed (since you inspect the first element in the container), which is a valid use case in my opinion. The problem is fixed by Richard's suggestion.